Added city Maurilia and corresponding data file and test

jerenner opened the pull request:
- The previous pull request was accidentally closed by deleting the branch
- The file NEW_se_mc_reco_pmaps_1evt.h5 is not written by the test but is data used to run the test.  The file NEW_se_mc_1evt is now written to a temporary directory in the test, as is the configuration file itself.
- The trailing underscores in for example  self.set_input_files_  are to avoid conflict with the corresponding function definition names

jjgomezcadenas committed:
Add new test suite for core/peak_functions

A new test suite (peak_functions_test.py) has been added. The module
runs several exahustive tests over peak_functions, including:

1. Comparison of BLR and CWF waveforms.
2. Comparison of python and cython functions (for this purpose the
python version of several existing cython functions have been added
to peak_functions.py)
3. Analysis of a toy model.

In addition, a few utility functions (needed for the test toy model) have
been added to core_functions (together with their corresponding test
in core_functions_test).

Some minor refactoring has been carried out along with the addition of this
module (moving an obsolete function out of peak_functions.py).

jacg committed:
Replace `toy_pmt_sum` with `toy_pmt_scale`

The original name, docstring and implementation were very misleading.

jacg committed:
Remove toy_pmt_scale altogether

Please decide whether keeping this function or inlining it at the only
place where it is used leads to more understandable test code. Pick
your favourite, perhaps add some explanatory comments and push a
separate commit with your choice on top. Don't worry about squashing
and rebasing, I'll do that.

jacg committed:
Enhance symmetry in test_csum_python_cython

Is there any good reason why the original Cython version used

    pmtblr[event].astype(np.float64)

while the Python version used

    pmtblr[event]

?

jacg committed:
Introduce a bug to motivate writing tests

jacg committed:
Fix obvious syntax errors

jacg committed:
Highlight symmetry

jjgomezcadenas committed:
Added two plotting functions

Functions to plot a signal (or the block of 12 PMT signals)
versus time (in mus) controlling time and signal intervals.

jjgomezcadenas committed:
Change in calibrated_pmt_sum: propagate to irene and to tests

calibrated_pmt_sum (cython version) returns now a tuple with the calibrated
sum (csum) computed without MAU threshold, and the csum computed with
MAU (previously it returned only the MAU-CSUM). The python version
returns the csum, csum-MAU and the MAU themselves (for the 12 pmts).

The change of interface is propagated to irene and to the various test functions.

jacg committed:
Cosmetics

jjgomezcadenas committed:
Improved function read_pmaps()

Function takes now the full path to the file and opens the file using
with context.

Commit:
Modified Maurilia to derive from base class

jjgomezcadenas committed:
Fixed bugs in several functions in pmaps

jacg commented on invisible_cities/cities/maurilia_test.py:
```diff
@@ -0,0 +1,71 @@
+
+config_file_contents = config_file_format.format(**config_file_spec)
+
+@mark.serial
```
The `serial` mark should be used in tests that cannot be run concurrently with other tests. The only reason for this that we have identified so far in IC is those tests which form a chain where some test uses files produced by another one.

I do not see why this test should be marked as serial.

jacg committed:
Introduce City and DeconvolutionCity base classes

jacg committed:
Move City bases into cities/base_cities.py

jacg committed:
Move input file setting to City base class

jacg committed:
Refactor Irene (with interface change)

jacg committed:
Cater for Python 2

Python 2 does not allow keyword arguments after **.

jacg committed:
Refactor Irene.store_pmaps()

jacg committed:
Adde test_command_line_irene

Discovered bug in utility functions which were using event without it
being in scope.

Added test configuration file creation in irene_test.py. Must look
into reducing duplication of config file creation in tests across
different cities.

jacg committed:
Refactor Diomira and add command line test
